test/functional/forall/atomic/tests/test-forall-atomic-basic.hpp

```diff
@@ -25,10 +27,11 @@ class ForallAtomicBasicFunctionalTest : public ::testing::Test
 template <typename ExecPolicy,
           typename AtomicPolicy,
           typename WORKINGRES,
+          typename SegmentType,
           typename T>
 void testAtomicFunctionBasic( RAJA::Index_type seglimit )
 {
-  RAJA::TypedRangeSegment<RAJA::Index_type> seg(0, seglimit);
+  SegmentType seg = RSMultiplexer<RAJA::Index_type, SegmentType>().makeseg(seglimit);
```

Comment:
Should this and seglimit be using type T instead of Index_type?

Reply:
This "seg" is a range segment, and seglimit should be either int or RAJA::Index_type to express the limit of the range. "T" can be a float or double and we probably don't want to pass a float/double into a rangesegment constructor.